### PR TITLE
Bring Metastore image up to date with features

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

File: hive-metastore/files/opt/bin/start-hive-metastore.sh [hunk 1]
```diff
@@ -36,18 +44,37 @@ sed -i \
   -e "s|%S3_ENDPOINT%|${S3_ENDPOINT:-}|g" \
   -e "s|%S3_ACCESS_KEY%|${S3_ACCESS_KEY:-}|g" \
   -e "s|%S3_SECRET_KEY%|${S3_SECRET_KEY:-}|g" \
+  -e "s|%REGION%|${REGION:-}|g" \
```
Comment (sopel39) [inline hive-metastore/files/opt/bin/start-hive-metastore.sh]: @kokosing why do we need `:-`? In AMI we don't have that

File: hive-metastore/files/opt/bin/hive-cli.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+test -v HIVE_METASTORE_KERBEROS_PRINCIPAL
+test -v GOOGLE_CLOUD_KEY_FILE_PATH
+test -v HIVE_METASTORE_URI
+
+sed -i \
+  -e "s|%S3_ENDPOINT%|${S3_ENDPOINT:-}|g" \
+  -e "s|%S3_ACCESS_KEY%|${S3_ACCESS_KEY:-}|g" \
+  -e "s|%S3_SECRET_KEY%|${S3_SECRET_KEY:-}|g" \
+  -e "s|%REGION%|${REGION:-}|g" \
+  -e "s|%HIVE_METASTORE_KERBEROS_PRINCIPAL%|${HIVE_METASTORE_KERBEROS_PRINCIPAL:-}|g" \
```
Comment (sopel39) [inline hive-metastore/files/opt/bin/hive-cli.sh]: @kokosing is empty principal OK?

File: hive-metastore/files/etc/hive/conf/hive-site.xml [hunk 3]
```diff
@@ -109,7 +124,7 @@
         </description>
     </property>
 
-	<property>
+    <property>
```
Comment (kokosing) [inline hive-metastore/files/etc/hive/conf/hive-site.xml]: separate commit?

File: hive-metastore/files/opt/bin/start-hive-metastore.sh (hunk 1 above)
Comment (kokosing) [reply hive-metastore/files/opt/bin/start-hive-metastore.sh]: If region is not given then, as I understand, instance region will be used.

File: hive-metastore/files/opt/bin/hive-cli.sh (hunk 2 above)
Comment (kokosing) [reply hive-metastore/files/opt/bin/hive-cli.sh]: I do not think so. How that could work? If principal is empty then we also have to disable SASL in hive-site.xml (unkerberize hive metastore).

File: hive-metastore/files/opt/bin/start-hive-metastore.sh [hunk 4]
```diff
@@ -36,18 +44,37 @@ sed -i \
...
   -e "s|%HIVE_METASTORE_KERBEROS_PRINCIPAL%|${HIVE_METASTORE_KERBEROS_PRINCIPAL:-}|g" \
+  -e "s|%GOOGLE_CLOUD_KEY_FILE_PATH%|${GOOGLE_CLOUD_KEY_FILE_PATH:-}|g" \
   /etc/hive/conf/hive-site.xml
 
+retry() {
+    wait_timeout="$1"
```
Comment (kokosing) [inline hive-metastore/files/opt/bin/start-hive-metastore.sh]: can you inline 300 here, and 1 below

File: hive-metastore/files/opt/bin/start-hive-metastore.sh [hunk 5]
```diff
@@ -74,4 +102,4 @@ else
 fi
 
 # log threshold is set to INFO to avoid log pollution from Datanucleus
-hive --service metastore --hiveconf hive.root.logger=INFO,console --hiveconf hive.log.threshold=INFO
+exec hive --service metastore --hiveconf hive.root.logger=INFO,console --hiveconf hive.log.threshold=INFO
```
Comment (kokosing) [inline hive-metastore/files/opt/bin/start-hive-metastore.sh]: separate commit?

File: hive-metastore/files/opt/bin/start-hive-metastore.sh [hunk 6]
```diff
@@ -59,9 +86,10 @@ elif [[ "$HIVE_METASTORE_DRIVER" == org.postgresql.Driver ]]; then
         export PGPASSWORD="$HIVE_METASTORE_PASSWORD"
         psql --host="$HIVE_METASTORE_DB_HOST" --username="$HIVE_METASTORE_USER" "$HIVE_METASTORE_DB_NAME" "$@"
     }
+    declare -fxr sql
 
     # Make sure that postgres is accessible
-    sql -c 'SELECT 1'
+    retry 300 1 timeout 5 bash -ce "sql -c 'SELECT 1'"
```
Comment (kokosing) [inline hive-metastore/files/opt/bin/start-hive-metastore.sh]: separate commit? Also we are already retrying in docker-compose.yml. Can you retry outside of this or should we remove retry from docker-compose.yml as well?

File: hive-metastore/files/opt/bin/start-hive-metastore.sh [hunk 7]
```diff
@@ -36,18 +44,37 @@ sed -i \
...
 export HIVE_METASTORE_DB_HOST="$(echo "$HIVE_METASTORE_JDBC_URL" | cut -d / -f 3 | cut -d : -f 1)"
 export HIVE_METASTORE_DB_NAME="$(echo "$HIVE_METASTORE_JDBC_URL" | cut -d / -f 4)"
 if [[ "$HIVE_METASTORE_DRIVER" == com.mysql.jdbc.Driver ]]; then
     sqlDir=/opt/sql/mysql
     function sql() {
         mysql --host="$HIVE_METASTORE_DB_HOST" --user="$HIVE_METASTORE_USER" --password="$HIVE_METASTORE_PASSWORD" "$HIVE_METASTORE_DB_NAME" "$@"
     }
-    # Make sure that postgres is accessible
-    sql -e 'SELECT 1'
+    declare -fxr sql
```
Comment (kokosing) [inline hive-metastore/files/opt/bin/start-hive-metastore.sh]: what is this? What is the purpose of this?

**Pull request closed without merging**